File: spec/coffeescripts/globalAnnouncementsSpec.js
/*
 * Copyright (C) 2016 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import $ from 'jquery'
import fakeENV from 'helpers/fakeENV'
import globalAnnouncements from 'global_announcements'
import serviceRCELoader from 'jsx/shared/rce/serviceRCELoader'

QUnit.module('GlobalAnnouncements', {
  setup() {
    fakeENV.setup()
    ENV.RICH_CONTENT_SERVICE_ENABLED = true
    this.loadOnTargetStub = sinon.stub(serviceRCELoader, 'loadOnTarget')
  },
  teardown() {
    serviceRCELoader.loadOnTarget.restore()
    $('#fixtures').empty()
    fakeENV.teardown()
  }
})

test('loads an editor for every matching node', function() {
  const html =
    "<textarea id='a1' class='edit_notification_form'>Announcement 1</textarea>" +
    "<textarea id='a2' class='edit_notification_form'>Announcement 2</textarea>" +
    "<form id='add_notification_form'><textarea id='a3'></textarea></form>"
  $(html).appendTo('#fixtures')
  globalAnnouncements.augmentView()
  ok(this.loadOnTargetStub.calledThrice)
})

File: spec/selenium/announcements/global_announcements_spec.rb
# frozen_string_literal: true

#
# Copyright (C) 2011 - present Instructure, Inc.
#
# This file is part of Canvas.
#
# Canvas is free software: you can redistribute it and/or modify it under
# the terms of the GNU Affero General Public License as published by the Free
# Software Foundation, version 3 of the License.
#
# Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
# details.
#
# You should have received a copy of the GNU Affero General Public License along
# with this program. If not, see <http://www.gnu.org/licenses/>.

require_relative '../common'

describe "global announcements" do
  include_context "in-process server selenium tests"

  before :each do
    course_with_student_logged_in
  end

  it 'shows empty message for both tabs', ignore_js_errors: true do
    get '/account_notifications'
    expect(fj("#tab-currentTab[aria-selected='true']:contains('Current')")).to be
    expect(fj("span:contains('Active Announcements')")).to be_displayed
    expect(fj("span:contains('No announcements to display')")).to be_displayed
    past_tab = f("#tab-pastTab")
    past_tab.click
    expect(fj("span:contains('Announcements from the past four months')")).to be_displayed
    expect(fj("span:contains('No announcements to display')")).to be_displayed
  end

  it 'shows notifications', ignore_js_errors: true do
    account_notification(:start_at => 2.days.ago, :end_at => 5.days.from_now, :send_message => true)
    account_notification(:message => 'from the past', :start_at => 6.days.ago, :end_at => 5.days.ago, :send_message => true)
    get '/account_notifications'
    expect(fj("h2:contains('this is a subject')")).to be_displayed
    expect(fj("span:contains('hi there')")).to be_displayed
    f("#tab-pastTab").click
    expect(fj("span:contains('from the past')")).to be_displayed
  end
end
